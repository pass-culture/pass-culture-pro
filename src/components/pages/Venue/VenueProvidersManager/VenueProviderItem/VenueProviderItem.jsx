import { Icon, pluralize } from 'pass-culture-shared'
import React from 'react'
import { NavLink } from 'react-router-dom'
import PropTypes from 'prop-types'
import { PROVIDER_ICONS } from './utils/providerIcons'

const VenueProviderItem = ({venueProvider}) => {
  const {lastSyncDate, nOffers, provider, venueIdAtOfferProvider, venueId} = venueProvider
  const {name: providerName, localClass} = provider
  const providerIcon = PROVIDER_ICONS[localClass]

  return (
    <li className="venue-provider-row">
      <Icon svg={providerIcon} />
      <div className="has-text-weight-bold fs14 provider-name-container">
        {providerName}
      </div>

      <div className="fs14 venue-id-at-offer-provider-container">
        <div>
          Compte : {' '}
          <strong className="fs14 has-text-weight-bold">
            {venueIdAtOfferProvider}
          </strong>

          {!lastSyncDate && (
            <div className="fs14 import-label-container">
              Importation en cours. Cette Ã©tape peut durer plusieurs dizaines de minutes.
            </div>
          )}
        </div>
      </div>

      <div className="offers-container">
        {lastSyncDate && (
          nOffers ? (
            <NavLink
              className="has-text-primary"
              to={`/offres?lieu=${venueId}`}
            >
              <Icon svg="ico-offres-r" width="22px" height="22px" />
              <div className="number-of-offers-label">
                {pluralize(nOffers, 'offres')}
              </div>
            </NavLink>
          ) : (
            <div className="number-of-offers-label">
              0 offre
            </div>
          )
        )}
      </div>
    </li>
  )
}

VenueProviderItem.propTypes = {
  venueProvider: PropTypes.object
}

export default VenueProviderItem
