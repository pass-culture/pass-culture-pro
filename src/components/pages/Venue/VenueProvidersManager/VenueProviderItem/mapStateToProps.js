import selectEventsByProviderId from 'selectors/selectEventsByProviderId'
import selectThingsByProviderId from 'selectors/selectThingsByProviderId'

function mapStateToProps(state, ownProps) {
  const {
    venueProvider: { providerId },
  } = ownProps
  return {
    events: selectEventsByProviderId(state, providerId),
    things: selectThingsByProviderId(state, providerId),
  }
}

export default mapStateToProps
