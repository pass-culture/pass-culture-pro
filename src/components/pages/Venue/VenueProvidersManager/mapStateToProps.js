import get from 'lodash.get'
import { selectCurrentUser } from 'with-login'

import selectProviderById from 'selectors/selectProviderById'
import selectProviders from 'selectors/selectProviders'
import selectVenueProviderByVenueIdAndVenueProviderId from './selectVenueProviderByVenueIdAndVenueProviderId'
import selectVenueProvidersByVenueId from './selectVenueProvidersByVenueId'

const mapStateToProps = (state, ownProps) => {
  const { venue } = ownProps
  const { id: venueId } = venue || {}
  const providers = selectProviders(state)

  const formPatch = get(state, 'form.venueProvider')

  let provider
  if (providers.length === 1) {
    provider = providers[0]
  } else {
    const providerId = get(formPatch, 'providerId')
    provider = selectProviderById(state, providerId)
  }

  return {
    currentUser: selectCurrentUser(state),
    provider,
    providers,
    venueProvider: selectVenueProviderByVenueIdAndVenueProviderId(
      state,
      venueId
    ),
    venueProviders: selectVenueProvidersByVenueId(state, venueId),
  }
}

export default mapStateToProps
