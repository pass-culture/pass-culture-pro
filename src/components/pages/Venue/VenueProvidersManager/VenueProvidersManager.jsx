import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import VenueProviderItem from './VenueProviderItem/VenueProviderItem'
import {
  ALLOCINE_PROVIDER_OPTION,
  DEFAULT_PROVIDER_OPTION,
  LIBRAIRES_PROVIDER_OPTION,
  TITELIVE_PROVIDER_OPTION,
} from './utils/providerOptions'
import AllocineProviderForm from './AllocineProviderForm/AllocineProviderFormContainer'
import TiteliveProviderForm from './TiteliveProviderForm/TiteliveProviderFormContainer'
import ReactTooltip from 'react-tooltip'
import LibrairesProviderForm from './LibrairesProviderForm/LibrairesProviderFormContainer'

class VenueProvidersManager extends PureComponent {
  constructor(props) {
    super(props)
    this.state = {
      isCreationMode: false,
      providerId: null,
      providerSelectedIsAllocine: false,
      providerSelectedIsTitelive: false,
      providerSelectedIsLibraires: false,
      venueIdAtOfferProviderIsRequired: true,
    }
  }

  componentDidMount() {
    const { loadProvidersAndVenueProviders } = this.props
    loadProvidersAndVenueProviders()
  }

  componentDidUpdate(prevProps) {
    ReactTooltip.rebuild()
    const { venueProviders } = this.props
    if (prevProps.venueProviders.length < venueProviders.length) {
      this.setState({ isCreationMode: false })
    }
  }

  handleAddVenueProvider = () => {
    this.setState({
      isCreationMode: true,
    })
  }

  handleChange = event => {
    const valueFromSelectInput = event.target.value
    const valueParsed = JSON.parse(valueFromSelectInput)
    this.setState({
      providerSelectedIsAllocine: false,
      providerSelectedIsTitelive: false,
      providerSelectedIsLibraires: false,
    })

    if (valueParsed && valueParsed.name === ALLOCINE_PROVIDER_OPTION.name) {
      this.setState({
        providerSelectedIsAllocine: true,
        venueIdAtOfferProviderIsRequired: valueParsed.requireProviderIdentifier,
      })
    } else if (valueParsed && valueParsed.name === TITELIVE_PROVIDER_OPTION.name) {
        this.setState({
          providerSelectedIsTitelive: true,
        })
    } else if (valueParsed && valueParsed.name === LIBRAIRES_PROVIDER_OPTION.name) {
        this.setState({
          providerSelectedIsLibraires: true,
        })
    }

    this.setState({
      providerId: valueParsed.id,
    })
  }

  render() {
    const { providers, match, venueProviders, venueSiret } = this.props
    const {
      isCreationMode,
      providerId,
      providerSelectedIsAllocine,
      providerSelectedIsTitelive,
      providerSelectedIsLibraires,
      venueIdAtOfferProviderIsRequired,
    } = this.state
    const hasAtLeastOneProvider = providers.length > 0
    const hasNoVenueProvider = venueProviders.length === 0

    return (
      <div className="venue-providers-manager section">
        <h2 className="main-list-title">
          {'Importation dâ€™offres'}
        </h2>

        <ul className="provider-list">
          {venueProviders.map(venueProvider => (
            <VenueProviderItem
              key={venueProvider.id}
              venueProvider={venueProvider}
            />
          ))}

          {isCreationMode && (
            <li className="add-provider-form">
              <div className="select-provider-section">
                <div id="select-source">
                  <label htmlFor="provider-options">
                    {'Source'}
                  </label>
                  <select
                    className="field-select"
                    id="provider-options"
                    onChange={this.handleChange}
                  >
                    <option
                      key={DEFAULT_PROVIDER_OPTION.id}
                      value={JSON.stringify(DEFAULT_PROVIDER_OPTION)}
                    >
                      {DEFAULT_PROVIDER_OPTION.name}
                    </option>
                    {providers.map(provider => {
                      return (
                        <option
                          key={`provider-${provider.id}`}
                          value={JSON.stringify(provider)}
                        >
                          {provider.name}
                        </option>
                      )
                    })}
                  </select>
                </div>
              </div>
              <div className="provider-form">
                {providerSelectedIsAllocine && (
                  <AllocineProviderForm
                    offererId={match.params.offererId}
                    providerId={providerId}
                    venueId={match.params.venueId}
                    venueIdAtOfferProviderIsRequired={venueIdAtOfferProviderIsRequired}
                  />
                )}

                {providerSelectedIsTitelive && (
                  <TiteliveProviderForm
                    offererId={match.params.offererId}
                    providerId={providerId}
                    venueId={match.params.venueId}
                    venueSiret={venueSiret}
                  />
                )}

                {providerSelectedIsLibraires && (
                  <LibrairesProviderForm
                    offererId={match.params.offererId}
                    providerId={providerId}
                    venueId={match.params.venueId}
                    venueSiret={venueSiret}
                  />
                )}
              </div>
            </li>
          )}
        </ul>

        {hasAtLeastOneProvider && hasNoVenueProvider && !isCreationMode && (
          <div className="has-text-centered">
            <button
              className="button is-secondary"
              id="add-venue-provider-btn"
              onClick={this.handleAddVenueProvider}
              type="button"
            >
              {'+ Importer des offres'}
            </button>
          </div>
        )}
      </div>
    )
  }
}

VenueProvidersManager.propTypes = {
  loadProvidersAndVenueProviders: PropTypes.func.isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape(),
  }).isRequired,
  providers: PropTypes.arrayOf(PropTypes.shape()).isRequired,
  venueProviders: PropTypes.arrayOf(PropTypes.shape()).isRequired,
  venueSiret: PropTypes.string.isRequired,
}

export default VenueProvidersManager
