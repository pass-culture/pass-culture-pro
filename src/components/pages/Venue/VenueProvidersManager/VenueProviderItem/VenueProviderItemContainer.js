import { connect } from 'react-redux'

import VenueProviderItem from './VenueProviderItem'
import mapStateToProps from './mapStateToProps'

export default connect(mapStateToProps)(VenueProviderItem)
