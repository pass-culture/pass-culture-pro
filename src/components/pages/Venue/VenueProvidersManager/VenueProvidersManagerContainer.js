import { connect } from 'react-redux'
import { withRouter } from 'react-router'
import { compose } from 'redux'
import { requestData } from 'redux-saga-data'

import VenueProvidersManager from './VenueProvidersManager'
import selectVenueProvidersByVenueId from './selectors/selectVenueProvidersByVenueId'
import { selectProviders } from '../../../../selectors/data/providersSelectors'

export const mapStateToProps = (state, ownProps) => {
  const { venue } = ownProps
  const { id: venueId, siret: venueSiret } = venue
  const providers = selectProviders(state)
  const venueProviders = selectVenueProvidersByVenueId(state, venueId)

  return {
    providers,
    venueProviders,
    venueSiret
  }
}

export const mapDispatchToProps = (dispatch, ownProps) => {
  return {
    loadProvidersAndVenueProviders: () => {
      const {
        match: {
          params: { venueId },
        },
      } = ownProps
      dispatch(
        requestData({
          apiPath: `/providers/${venueId}`,
        })
      )
      dispatch(
        requestData({
          apiPath: `/venueProviders?venueId=${venueId}`,
        })
      )
    },
  }
}

export default compose(
  withRouter,
  connect(
    mapStateToProps,
    mapDispatchToProps
  )
)(VenueProvidersManager)
