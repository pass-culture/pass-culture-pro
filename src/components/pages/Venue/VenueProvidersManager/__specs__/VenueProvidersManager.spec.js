import { shallow } from 'enzyme'
import React from 'react'
import VenueProvidersManager, { FormRendered } from '../VenueProvidersManager'
import { Field, Form } from 'react-final-form'
import { HiddenField, TextField } from '../../../../layout/form/fields'
import { Icon } from '../../../../layout/Icon'
import VenueProviderItem from '../VenueProviderItem/VenueProviderItem'

describe('src | components | pages | Venue | VenueProvidersManager', () => {
  let props
  let dispatch
  let history

  beforeEach(() => {
    dispatch = jest.fn()
    history = {
      push: jest.fn()
    }
    props = {
      dispatch,
      history,
      match: {
        params: {
          offererId: 'CC',
          venueId: 'AB',
          venueProviderId: 'AE'
        }
      },
      providers: [
        {id: 'DD', requireProviderIdentifier: true},
        {id: 'EE', requireProviderIdentifier: true}
      ],
      venue: {id: 'AB'},
      venueProviders: [{id: 'AA'}, {id: 'BB'}],
    }
  })

  describe('snapshot', () => {
    it('should match snapshot', () => {
      // when
      const wrapper = shallow(<VenueProvidersManager {...props} />)

      // then
      expect(wrapper).toMatchSnapshot()
    })

    it('should return is creation mode to true when query param "venueProviderId" is "nouveau"', () => {
      // given
      props.match.params.venueProviderId = 'nouveau'

      // when
      const wrapper = shallow(<VenueProvidersManager {...props} />)

      // then
      expect(wrapper.state()).toEqual({
        isCreationMode: true,
        isLoadingMode: false,
        isProviderSelected: false,
        venueIdAtOfferProviderIsRequired: true
      })
    })
  })

  describe('render', () => {
    it('should initialize VenueProvidersManager component with default state', () => {
      // when
      const wrapper = shallow(<VenueProvidersManager {...props} />)

      // then
      expect(wrapper.state()).toEqual({
        isCreationMode: false,
        isLoadingMode: false,
        isProviderSelected: false,
        venueIdAtOfferProviderIsRequired: true
      })
    })

    it('should contain a block with information regarding importation process', () => {
      // when
      const wrapper = shallow(<VenueProvidersManager {...props} />)

      // then
      const title = wrapper.find('h2')
      expect(title).toBeDefined()
      const span = title.find('span')
      expect(span.text()).toBe('Si vous avez plusieurs comptes auprès de la même source, ajoutez-les successivement.')
    })

    it('should display 2 VenueProviderItemContainer when there are 2 venue providers', () => {
      // when
      const wrapper = shallow(<VenueProvidersManager {...props} />)

      // then
      const venueProviderItemContainers = wrapper.find(VenueProviderItem)
      expect(venueProviderItemContainers).toHaveLength(2)
      expect(venueProviderItemContainers.at(0).prop('venueProvider')).toEqual({id: 'AA'})
      expect(venueProviderItemContainers.at(1).prop('venueProvider')).toEqual({id: 'BB'})
    })

    it('should retrieve providers and venue providers when component is mounted', () => {
      // when
      shallow(<VenueProvidersManager {...props} />)

      // then
      expect(dispatch.mock.calls[0][0]).toEqual({
        config: {
          apiPath: '/providers',
          method: 'GET'
        },
        type: 'REQUEST_DATA_GET_/PROVIDERS'
      })
      expect(dispatch.mock.calls[1][0]).toEqual({
        config: {
          apiPath: '/venueProviders?venueId=AB',
          method: 'GET'
        },
        type: 'REQUEST_DATA_GET_/VENUEPROVIDERS?VENUEID=AB'
      })
    })

    it('should update current URL when clicking on add offer button', async () => {
      // given
      const wrapper = shallow(<VenueProvidersManager {...props} />)
      const addOfferBtn = wrapper.find('#add-offer-btn')

      // when
      addOfferBtn.simulate('click')

      // then
      expect(history.push).toHaveBeenCalledWith('/structures/CC/lieux/AB/fournisseurs/nouveau')
    })
  })

  describe('functions', () => {
    describe('handleSubmit', () => {
      it('should update venue provider using API', () => {
        // given
        const formValues = {
          id: 'AA',
          provider: "{\"id\": \"CC\"}",
          venueIdAtOfferProvider: 'BB'
        }
        const wrapper = shallow(<VenueProvidersManager {...props} />)

        // when
        wrapper.instance().handleSubmit(formValues)

        // then
        expect(wrapper.state('isLoadingMode')).toBe(true)
        expect(dispatch).toHaveBeenCalledWith({
          config: {
            apiPath: '/venueProviders',
            body: {
              providerId: 'CC',
              venueId: 'AA',
              venueIdAtOfferProvider: 'BB'
            },
            handleFail: expect.any(Function),
            handleSuccess: expect.any(Function),
            method: 'POST'
          },
          type: 'REQUEST_DATA_POST_/VENUEPROVIDERS'
        })
      })
    })

    describe('handleSuccess', () => {
      it('should update current url when action was handled successfully', () => {
        // given
        const wrapper = shallow(<VenueProvidersManager {...props} />)

        // when
        wrapper.instance().handleSuccess()

        // then
        expect(history.push).toHaveBeenCalledWith('/structures/CC/lieux/AB')
      })
    })

    describe('handleFail', () => {
      it('should display a notification with the proper informations', () => {
        // given
        const wrapper = shallow(<VenueProvidersManager {...props} />)
        const action = {
          payload: {
            errors: [{
              error: 'fake error'
            }]
          }
        }

        // when
        wrapper.instance().handleFail({}, action)

        // then
        expect(dispatch).toHaveBeenCalledWith({
          patch: {
            text: 'fake error',
            type: 'fail'
          },
          type: 'SHOW_NOTIFICATION'
        })
      })

      it('should reset component state with the default values', () => {
        // given
        const wrapper = shallow(<VenueProvidersManager {...props} />)
        const action = {
          payload: {
            errors: [{
              error: 'fake error'
            }]
          }
        }

        // when
        wrapper.instance().handleFail({}, action)

        // then
        expect(wrapper.state()).toEqual({
          isCreationMode: false,
          isLoadingMode: false,
          isProviderSelected: false,
          venueIdAtOfferProviderIsRequired: true
        })
      })
    })

    describe('handleChange', () => {
      let input
      let onChange

      beforeEach(() => {
        onChange = jest.fn()
        input = {
          onChange
        }
      })

      it('should update state values when selected option is valid and different from default value', () => {
        // given
        const event = {
          target: {
            value: "{\"id\":\"AE\", \"requireProviderIdentifier\": true}"
          }
        }
        const wrapper = shallow(<VenueProvidersManager {...props} />)

        // when
        wrapper.instance().handleChange(event, input)

        // then
        expect(wrapper.state('isProviderSelected')).toBe(true)
        expect(wrapper.state('venueIdAtOfferProviderIsRequired')).toBe(true)
      })

      it('should reset form state when selected option is equal to default value', () => {
        // given
        const event = {
          target: {
            value: "{\"id\":\"default\", \"name\": \"Choix de la source\"}"
          }
        }
        const wrapper = shallow(<VenueProvidersManager {...props} />)

        // when
        wrapper.instance().handleChange(event, input)

        // then
        expect(wrapper.state()).toEqual({
          isCreationMode: false,
          isLoadingMode: false,
          isProviderSelected: false,
          venueIdAtOfferProviderIsRequired: true
        })
      })
    })
  })

  describe('form', () => {
    beforeEach(() => {
      props.match.params.venueProviderId = 'nouveau'
      props.venueProviders = []
    })

    it('should render form with the proper values using user input', async () => {
      // when
      const wrapper = shallow(<VenueProvidersManager {...props} />)

      // then
      const form = wrapper.find(Form)
      expect(form).toBeDefined()
      expect(form.prop('decorators')).toEqual(expect.anything())
      expect(form.prop('initialValues')).toEqual({id: 'AB'})
      expect(form.prop('onSubmit')).toEqual(expect.any(Function))
      expect(form.prop('render')).toEqual(expect.any(Function))
    })

    describe('FormRendered component', () => {
      let props
      let handleSubmit
      let handleChange

      beforeEach(() => {
        handleSubmit = jest.fn()
        handleChange = jest.fn()
        props = {
          handleChange,
          providers: [{id: 'AA'}, {id: 'BB'}],
          isProviderSelected: false,
          isLoadingMode: false,
          isCreationMode: false,
          venueIdAtOfferProviderIsRequired: false,
        }
      })

      it('should render a HiddenField component with the right props', () => {
        // when
        const renderedForm = shallow(FormRendered({...props})(handleSubmit))

        // then
        const hiddenField = renderedForm.find(HiddenField)
        expect(hiddenField).toBeDefined()
        expect(hiddenField.prop('name')).toBe('id')
      })

      it('should render a Field component as select input with providers as options when provided', () => {
        // when
        const renderedForm = shallow(FormRendered({...props})(handleSubmit))

        // then
        const selectField = renderedForm.find(Field)
        expect(selectField).toHaveLength(1)
        expect(selectField.prop('name')).toBe('provider')
        expect(selectField.prop('required')).toBe(true)
      })

      it('should render a TextField component not in read only mode when provider is selected, not in loading mode and provider identifier is required', () => {
        // given
        props.isProviderSelected = true
        props.isLoadingMode = false
        props.venueIdAtOfferProviderIsRequired = true

        // when
        const renderedForm = shallow(FormRendered({...props})(handleSubmit))

        // then
        const textField = renderedForm.find(TextField)
        expect(textField).toHaveLength(1)
        expect(textField.prop('className')).toBe('field-text fs12')
        expect(textField.prop('label')).toBe('Compte : ')
        expect(textField.prop('name')).toBe('venueIdAtOfferProvider')
        expect(textField.prop('readOnly')).toBe(false)
        expect(textField.prop('required')).toBe(true)
      })

      it('should render a TextField component in read only mode when provider is selected, in loading mode and provider identifier is required', () => {
        // given
        props.isProviderSelected = true
        props.isLoadingMode = true
        props.venueIdAtOfferProviderIsRequired = true

        // when
        const renderedForm = shallow(FormRendered({...props})(handleSubmit))

        // then
        const textField = renderedForm.find(TextField)
        expect(textField).toHaveLength(1)
        expect(textField.prop('className')).toBe('field-text fs12 field-is-read-only')
        expect(textField.prop('label')).toBe('Compte : ')
        expect(textField.prop('name')).toBe('venueIdAtOfferProvider')
        expect(textField.prop('readOnly')).toBe(true)
        expect(textField.prop('required')).toBe(true)
      })

      it('should render a TextField component in read only mode when provider is selected, not loading mode and provider identifier is not required', () => {
        // given
        props.isProviderSelected = true
        props.isLoadingMode = true
        props.venueIdAtOfferProviderIsRequired = false

        // when
        const renderedForm = shallow(FormRendered({...props})(handleSubmit))

        // then
        const textField = renderedForm.find(TextField)
        expect(textField).toHaveLength(1)
        expect(textField.prop('className')).toBe('field-text fs12 field-is-read-only')
        expect(textField.prop('label')).toBe('Compte : ')
        expect(textField.prop('name')).toBe('venueIdAtOfferProvider')
        expect(textField.prop('readOnly')).toBe(true)
        expect(textField.prop('required')).toBe(true)
      })

      it('should display a message information regarding data import when in loading mode', () => {
        // given
        props.isProviderSelected = true
        props.isLoadingMode = true

        // when
        const renderedForm = shallow(FormRendered({...props})(handleSubmit))

        // then
        const importLabelContainer = renderedForm.find('.import-label-container')
        const importLabelSpan = importLabelContainer.find('span')
        expect(importLabelContainer).toHaveLength(1)
        expect(importLabelSpan).toHaveLength(1)
        expect(importLabelSpan.text()).toBe('Importation en cours. Cette étape peut durer plusieurs dizaines de minutes.')
      })

      it('should display a tooltip and an Icon component when provider is selected, not in loading mode and provider identifier is required', () => {
        // given
        props.isProviderSelected = true
        props.isLoadingMode = false
        props.venueIdAtOfferProviderIsRequired = true

        // when
        const renderedForm = shallow(FormRendered({...props})(handleSubmit))

        // then
        const tooltip = renderedForm.find('.tooltip-info')
        expect(tooltip).toHaveLength(1)
        expect(tooltip.prop('className')).toBe('tooltip tooltip-info')
        expect(tooltip.prop('data-place')).toBe('bottom')
        expect(tooltip.prop('data-tip')).toBe('<p>Veuillez saisir un identifiant.</p>')
        const icon = tooltip.find(Icon)
        expect(icon).toHaveLength(1)
        expect(icon.prop('svg')).toBe('picto-info')
      })

      it('should not display a tooltip and an Icon component when provider is selected, not in loading mode and provider identifier is not required', () => {
        // given
        props.isProviderSelected = true
        props.isLoadingMode = false
        props.venueIdAtOfferProviderIsRequired = false

        // when
        const renderedForm = shallow(FormRendered({...props})(handleSubmit))

        // then
        const tooltip = renderedForm.find('.tooltip-info')
        expect(tooltip).toHaveLength(0)
      })

      it('should display an import button when provider is selected, in creation mode and not in loading mode', () => {
        // given
        props.isProviderSelected = true
        props.isCreationMode = true
        props.isLoadingMode = false

        // when
        const renderedForm = shallow(FormRendered({...props})(handleSubmit))

        // then
        const importButtonContainer = renderedForm.find('.provider-import-button-container')
        expect(importButtonContainer).toHaveLength(1)
        const importButton = importButtonContainer.find('button')
        expect(importButton).toHaveLength(1)
        expect(importButton.prop('className')).toBe('button is-intermediate provider-import-button')
        expect(importButton.prop('type')).toBe('submit')
        expect(importButton.text()).toBe('Importer')
      })
    })
  })
})
