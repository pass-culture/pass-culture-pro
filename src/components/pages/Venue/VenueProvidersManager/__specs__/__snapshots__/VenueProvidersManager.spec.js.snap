// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | Venue | VenueProvidersManager snapshot should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VenueProvidersManager
    dispatch={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "config": Object {
                "apiPath": "/providers",
                "method": "GET",
              },
              "type": "REQUEST_DATA_GET_/PROVIDERS",
            },
          ],
          Array [
            Object {
              "config": Object {
                "apiPath": "/venueProviders?venueId=AB",
                "method": "GET",
              },
              "type": "REQUEST_DATA_GET_/VENUEPROVIDERS?VENUEID=AB",
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    history={
      Object {
        "push": [MockFunction],
      }
    }
    match={
      Object {
        "params": Object {
          "offererId": "CC",
          "venueId": "AB",
          "venueProviderId": "AE",
        },
      }
    }
    providers={
      Array [
        Object {
          "id": "DD",
          "requireProviderIdentifier": true,
        },
        Object {
          "id": "EE",
          "requireProviderIdentifier": true,
        },
      ]
    }
    venue={
      Object {
        "id": "AB",
      }
    }
    venueProviders={
      Array [
        Object {
          "id": "AA",
        },
        Object {
          "id": "BB",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="main-list-title"
        >
          IMPORTATIONS D'OFFRES
          <span
            className="is-pulled-right is-size-7 has-text-grey"
          >
            Si vous avez plusieurs comptes auprès de la même source, ajoutez-les successivement.
          </span>
        </h2>,
        <ul
          className="main-list"
        >
          <VenueProviderItem
            venueProvider={
              Object {
                "id": "AA",
              }
            }
          />
          <VenueProviderItem
            venueProvider={
              Object {
                "id": "BB",
              }
            }
          />
        </ul>,
        <div
          className="has-text-centered"
        >
          <button
            className="button is-secondary"
            disabled={false}
            id="add-offer-btn"
            onClick={[Function]}
            type="button"
          >
            + Importer des offres
          </button>
        </div>,
      ],
      "className": "venue-providers-manager section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "IMPORTATIONS D'OFFRES",
            <span
              className="is-pulled-right is-size-7 has-text-grey"
            >
              Si vous avez plusieurs comptes auprès de la même source, ajoutez-les successivement.
            </span>,
          ],
          "className": "main-list-title",
        },
        "ref": null,
        "rendered": Array [
          "IMPORTATIONS D'OFFRES",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Si vous avez plusieurs comptes auprès de la même source, ajoutez-les successivement.",
              "className": "is-pulled-right is-size-7 has-text-grey",
            },
            "ref": null,
            "rendered": "Si vous avez plusieurs comptes auprès de la même source, ajoutez-les successivement.",
            "type": "span",
          },
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            Array [
              <VenueProviderItem
                venueProvider={
                  Object {
                    "id": "AA",
                  }
                }
              />,
              <VenueProviderItem
                venueProvider={
                  Object {
                    "id": "BB",
                  }
                }
              />,
            ],
            false,
          ],
          "className": "main-list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "AA",
            "nodeType": "class",
            "props": Object {
              "venueProvider": Object {
                "id": "AA",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "BB",
            "nodeType": "class",
            "props": Object {
              "venueProvider": Object {
                "id": "BB",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="button is-secondary"
            disabled={false}
            id="add-offer-btn"
            onClick={[Function]}
            type="button"
          >
            + Importer des offres
          </button>,
          "className": "has-text-centered",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "+ Importer des offres",
            "className": "button is-secondary",
            "disabled": false,
            "id": "add-offer-btn",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "+ Importer des offres",
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="main-list-title"
          >
            IMPORTATIONS D'OFFRES
            <span
              className="is-pulled-right is-size-7 has-text-grey"
            >
              Si vous avez plusieurs comptes auprès de la même source, ajoutez-les successivement.
            </span>
          </h2>,
          <ul
            className="main-list"
          >
            <VenueProviderItem
              venueProvider={
                Object {
                  "id": "AA",
                }
              }
            />
            <VenueProviderItem
              venueProvider={
                Object {
                  "id": "BB",
                }
              }
            />
          </ul>,
          <div
            className="has-text-centered"
          >
            <button
              className="button is-secondary"
              disabled={false}
              id="add-offer-btn"
              onClick={[Function]}
              type="button"
            >
              + Importer des offres
            </button>
          </div>,
        ],
        "className": "venue-providers-manager section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "IMPORTATIONS D'OFFRES",
              <span
                className="is-pulled-right is-size-7 has-text-grey"
              >
                Si vous avez plusieurs comptes auprès de la même source, ajoutez-les successivement.
              </span>,
            ],
            "className": "main-list-title",
          },
          "ref": null,
          "rendered": Array [
            "IMPORTATIONS D'OFFRES",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Si vous avez plusieurs comptes auprès de la même source, ajoutez-les successivement.",
                "className": "is-pulled-right is-size-7 has-text-grey",
              },
              "ref": null,
              "rendered": "Si vous avez plusieurs comptes auprès de la même source, ajoutez-les successivement.",
              "type": "span",
            },
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              Array [
                <VenueProviderItem
                  venueProvider={
                    Object {
                      "id": "AA",
                    }
                  }
                />,
                <VenueProviderItem
                  venueProvider={
                    Object {
                      "id": "BB",
                    }
                  }
                />,
              ],
              false,
            ],
            "className": "main-list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "AA",
              "nodeType": "class",
              "props": Object {
                "venueProvider": Object {
                  "id": "AA",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "BB",
              "nodeType": "class",
              "props": Object {
                "venueProvider": Object {
                  "id": "BB",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            false,
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="button is-secondary"
              disabled={false}
              id="add-offer-btn"
              onClick={[Function]}
              type="button"
            >
              + Importer des offres
            </button>,
            "className": "has-text-centered",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "+ Importer des offres",
              "className": "button is-secondary",
              "disabled": false,
              "id": "add-offer-btn",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "+ Importer des offres",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
